src/test/scala/chiseltest/formal/MagicPacketTrackerTests.scala

```diff
@@ -0,0 +1,383 @@
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    val e = intercept[FailedBoundedCheckException] {
+      verify(new QueueFormalTest(new MyQueueV2(DefaultDepth, 32)), defaultOptions)
+    }
+    assert(e.failAt == 3)
+  }
+
+  it should "verify QueueV3" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV3(DefaultDepthPow2,32)), defaultOptions)
+  }
+
+  it should "verify QueueV4" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV4(DefaultDepthPow2,32)), defaultOptions)
+  }
+
+  it should "verify QueueV5" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV5(DefaultDepthPow2,32)), defaultOptions)
+  }
+
+  it should "find bug in QueueV6 w/ pipe = false" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    val e = intercept[FailedBoundedCheckException] {
+      verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = false, fixed = false)), defaultOptions)
+    }
+    assert(e.failAt == 3)
+  }
+
+  it should "verify fix for QueueV6 w/ pipe = false" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = false, fixed = true)), defaultOptions)
+  }
+
+  it should "find bug in QueueV6 w/ pipe = true" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    val e = intercept[FailedBoundedCheckException] {
+      verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = true, fixed = false)), defaultOptions)
+    }
+    assert(e.failAt == 2)
+  }
+
+  it should "verify fix for QueueV6 w/ pipe = true" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = true, fixed = true)), defaultOptions)
+  }
+
+  it should "verify some versions of the Queue from the Chisel standard library" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new ChiselQueueTest(DefaultDepth, 32, false, false), defaultOptions)
+    verify(new ChiselQueueTest(DefaultDepth, 32, false, true), defaultOptions)
+    verify(new ChiselQueueTest(DefaultDepth, 32, true, false), defaultOptions)
+    verify(new ChiselQueueTest(DefaultDepth, 32, true, true), defaultOptions)
+  }
+}
+
+private class PacketTrackerCounterOverflowTest extends Module {
```

Comment:
What is this the point of this test? Is it to make sure the overflow assertion fires since that shouldn't happen in the other tests?

Reply:
Yes. Essentially what happens here is that the depth parameter of the tracker is set incorrectly which means that the internal counter cannot track all the elements that the queue can safe. In all other tests, the depth of the tracker is set correctly to equal the depth of the queue.